### PR TITLE
[Space Engine] Remove the @_downgrade_exhaustivity_check hack

### PR DESCRIPTION
**Pull request by CodaFi**
This hack was only needed for Swift 3 mode in a narrow case.  Flush it out of the compiler so we can simplify the space engine.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please test source compatibility

**Comment by davidungar** [comment]
Good stuff!

**Comment by CodaFi** [comment]
Please test with following pull request:
https://github.com/apple/swift-corelibs-libdispatch/pull/381

@swift-ci Please smoke test Linux platform

File: include/swift/AST/Attr.def [hunk 1]
```diff
@@ -346,12 +346,6 @@ DECL_ATTR(_restatedObjCConformance, RestatedObjCConformance,
   OnProtocol |
   LongAttribute | RejectByParser |
   NotSerialized, 70)
-// HACK: Attribute needed to preserve source compatibility by downgrading errors
-// due to an SDK change in Dispatch
-SIMPLE_DECL_ATTR(_downgrade_exhaustivity_check, DowngradeExhaustivityCheck,
-  OnEnumElement |
-  LongAttribute | UserInaccessible,
-  71)
```
Comment (jrose-apple) [inline include/swift/AST/Attr.def]: Can you leave behind a "71 is unused comment" here, like we did for other enums?

Technically this should probably mean bumping the module version too

File: lib/SILGen/SILGenPattern.cpp [hunk 2]
```diff
@@ -1812,8 +1812,7 @@ CaseBlocks::CaseBlocks(
                                SGF.F.getResilienceExpansion());
   }
   if (canAssumeExhaustive) {
-    // Check that Sema didn't let any cases slip through. (This can happen
-    // with @_downgrade_exhaustivity_check.)
+    // Check that Sema didn't let any cases slip through.
```
Comment (jrose-apple) [inline lib/SILGen/SILGenPattern.cpp]: I think this can't happen anymore, so maybe we can make this an assertion instead.

**Review by jrose-apple — COMMENTED** [review]
Looks good! Only small comments, not relevant to the main diff.

**Comment by CodaFi** [comment]
Lost the original source compat failure.  Let's try again

@swift-ci please test source compatibility

**Comment by CodaFi** [comment]
Please test with following pull request:
apple/swift-corelibs-libdispatch#381

@swift-ci Please smoke test

**Comment by CodaFi** [comment]
Please test with following pull request:
apple/swift-corelibs-libdispatch#381

@swift-ci Please smoke test Linux platform

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**